os/fs/driver/mtd/smart.c

```diff
@@ -2830,7 +2830,7 @@ static inline int smart_llformat(FAR struct smart_struct_s *dev, unsigned long a
 
 	/* Erase the MTD device. */
 
-	ret = MTD_IOCTL(dev->mtd, MTDIOC_BULKERASE, 0);
+	ret = MTD_IOCTL(dev->mtd, MTDIOC_BULKERASE, (unsigned long)((uintptr_t)&dev->geo));
```

Comment:
So, prefious is correct.
To sum up,
If mtd driver get MTDIOC_BULKERASE as a ioctl args, it means erase entire of flash.
But mtd_partition will receive ioctl before mtd driver receive it if system using mtd partition.
To prevent erase another partition, mtd_partiton calls erase instead of ioctl with MTDIOC_BULKERASE.
If there's no partition, mtd driver will erase entire of flash.

Reply:
You're right. Reverted.